src/extra/gpkg/read.jl

```diff
@@ -0,0 +1,172 @@
+#
+# Requirement 22: gpkg_geometry_columns table
+# SHALL contain one row record for the geometry column
+# in each vector feature data table
+#
+# Requirement 23: gpkg_geometry_columns table_name column
+# SHALL reference values in the gpkg_contents table_name column
+# for rows with a data_type of 'features'
+#
+# Requirement 24: The column_name column value in a gpkg_geometry_columns row
+# SHALL be the name of a column in the table or view specified by the table_name
+# column value for that row.
+#
+# Requirement 25: The geometry_type_name value in a gpkg_geometry_columns row
+# SHALL be one of the uppercase geometry type names specified
+#
+# Requirement 146: The srs_id value in a gpkg_geometry_columns table row
+# SHALL match the srs_id column value from the corresponding row in the
+# gpkg_contents table.
+function gpkgextract(db; layer=1)
+  # get the first (and only) feature table returned in sqlite query results
+  metadata = first(DBInterface.execute(
+    db,
+    """
+    SELECT g.table_name AS tablename, g.column_name AS geomcolumn, g.z as zextent,
+    c.srs_id AS srsid, srs.organization AS org, srs.organization_coordsys_id AS code
+    FROM gpkg_geometry_columns g, gpkg_spatial_ref_sys srs
+    JOIN gpkg_contents c ON ( g.table_name = c.table_name )
+    WHERE c.data_type = 'features'
+    AND g.srs_id = c.srs_id
+    LIMIT 1 OFFSET ($layer-1);
+    """
+  ))
+
+  # According to https://www.geopackage.org/spec/#r33, feature table geometry columns
+  # SHALL contain geometries with the srs_id specified for the column by the gpkg_geometry_columns table srs_id column value.
+  org = metadata.org
+  code = metadata.code
+  srsid = metadata.srsid
+  if srsid == 0 || srsid == 4326
+    crs = isone(metadata.zextent) ? LatLonAlt{WGS84Latest} : LatLon{WGS84Latest}
+  elseif srsid == -1
+    crs = Cartesian{NoDatum}
+  else
+    if org == "EPSG"
+      crs = CoordRefSystems.get(EPSG{code})
+    elseif org == "ESRI"
+      crs = CoordRefSystems.get(ESRI{code})
+    end
+  end
+
+  # According to https://www.geopackage.org/spec/#r14
+  # The table_name column value in a gpkg_contents table row 
+  # SHALL contain the name of a SQLite table or view.
+  tablename = metadata.tablename
+  geomcolumn = metadata.geomcolumn
+  tableinfo = SQLite.tableinfo(db, tablename)
+
+  # "pk" (either zero for columns that are not part of the primary key, or the 1-based index of the column within the primary key)
+  columns = [name for (name, pk) in zip(tableinfo.name, tableinfo.pk) if pk == 0]
+  gpkgbinary = DBInterface.execute(db, "SELECT  $(join(columns, ',')) FROM $tablename;")
+  table = map(gpkgbinary) do row
+    # According to https://www.geopackage.org/spec/#r30
+    # A feature table or view SHALL have only one geometry column.
+    geomindex = findfirst(==(Symbol(geomcolumn)), keys(row))
+    values = map(keys(row)[[begin:(geomindex - 1); (geomindex + 1):end]]) do key
+      key, getproperty(row, key)
+    end
```

Comment:
I asked a few times without response: can we use a simpler namedtuple syntax here? Shared an example in a previous comment...